port/linuxaarch64/omrsignal_context.c

```diff
@@ -0,0 +1,357 @@
+
+void
+fillInUnixSignalInfo(struct OMRPortLibrary *portLibrary, void *contextInfo, struct OMRUnixSignalInfo *j9Info)
+{
+	j9Info->platformSignalInfo.context = (ucontext_t *)contextInfo;
+	/* module info is filled on demand */
+}
+
+uint32_t
+infoForSignal(struct OMRPortLibrary *portLibrary, struct OMRUnixSignalInfo *info, int32_t index, const char **name, void **value)
+{
+	*name = "";
+
+	switch (index) {
+
+	case OMRPORT_SIG_SIGNAL_TYPE:
+	case 0:
+		*name = "J9Generic_Signal_Number";
+		*value = &info->portLibrarySignalType;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_PLATFORM_SIGNAL_TYPE:
+	case 1:
+		*name = "Signal_Number";
+		if( NULL != info->sigInfo) {
+			*value = &info->sigInfo->si_signo;
+			return OMRPORT_SIG_VALUE_32;
+		}
+		break;
+
+	case OMRPORT_SIG_SIGNAL_ERROR_VALUE:
+	case 2:
+		*name = "Error_Value";
+		*value = &info->sigInfo->si_errno;
+		return OMRPORT_SIG_VALUE_32;
+
+	case OMRPORT_SIG_SIGNAL_CODE:
+	case 3:
+		*name = "Signal_Code";
+		if( NULL != info->sigInfo) {
+			*value = &info->sigInfo->si_code;
+			return OMRPORT_SIG_VALUE_32;
+		}
+		break;
+
+	case OMRPORT_SIG_SIGNAL_HANDLER:
+	case 4:
+		*name = "Handler1";
+		*value = &info->handlerAddress;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+
+	case 5:
+		*name = "Handler2";
+		*value = &info->handlerAddress2;
+		return OMRPORT_SIG_VALUE_ADDRESS;
+	
+	case OMRPORT_SIG_SIGNAL_INACCESSIBLE_ADDRESS:
+	case 6:
+		/* si_code > 0 indicates that the signal was generated by the kernel */
+		if ( NULL != info->sigInfo && 0 < info->sigInfo->si_code ) {
+			if ((info->sigInfo->si_signo == SIGBUS) || (info->sigInfo->si_signo == SIGSEGV)) {
+				*name = "InaccessibleAddress";
+				*value = &info->sigInfo->si_addr;
+				return OMRPORT_SIG_VALUE_ADDRESS;
+			}
+		}
+		break;
+
+	// case OMRPORT_SIG_SIGNAL_ADDRESS:
+	// case 7:
+	// 	*name = "fault_address";
+	// 	*value = &((struct sigcontext *)&info->platformSignalInfo.context->uc_mcontext)->fault_address;
+	// 	return OMRPORT_SIG_VALUE_ADDRESS;
+
+
+	default:
+		break;
+	}
+	
+	return OMRPORT_SIG_VALUE_UNDEFINED;
+}
+
+
+/**
+ * Walk the 4k reserve space to get the floating point registers since the layout differ 
+ * per-chip
+ * @param	magic_to_find	input	magic number #define in the linux kernel (sigcontext.h)\
+ * that defines which struct the reserved area is at a given space
+ * @param	reserved_space	input	array of bytes that holds all the struct
+ * @see linux/arch/arm64/include/uapi/asm/sigcontext.h for the defines and information to why we need
+ * to walk the space
+
+ */
+void*
+walkReserveSpace(uint32_t magic_to_find, uint8_t *reserved_space)
+{
+	if ( NULL == reserved_space ) {
+		return NULL;
+	}
+	
+	int i = 0;
+	while ( RESERVED_SPACE_SZ > i ) {
+		struct _aarch64_ctx *header = (struct _aarch64_ctx *)&reserved_space[i];
+		if ( NULL == header || 0 < header->size ) {
+			return NULL;
+		} else if ( magic_to_find == header->magic ) {
+			return (void*)&reserved_space[i];
+		}
+		i += header->size;
```

Comment:
When reaching here, header->size should always be 0, correct ?

Reply:
typo (should be ==)
fixed! thanks for that! also no the reserved space could be shorter than 4096 bytes so the header size gives us the start of the next struct and help us wallk through it. since there is no contract anywhere of where the could store the floating point registers, we have to walk the whole space is my understanding (see doxygen comments on that function for more info)